frontend/ui/widget/textboxwidget.lua

```diff
@@ -1168,6 +1169,28 @@ function TextBoxWidget:update(scheduled_update)
     self.scheduled_update = nil
 end
 
+function TextBoxWidget:setText(text)
+    if text == self.text then
+        return
+    end
+
+    self.text = text
+
+    if self.use_xtext then
+        self:_measureWithXText()
+    else
+        self:_evalCharWidthList()
+    end
+    self:_splitToLines()
+
+    self:update()
+end
+dbg:guard(TextBoxWidget, "setText",
+    function(self, text)
+        assert(type(text) == "string",
+            "Wrong text type (expected string)")
+    end)
```

Comment:
What's this sneaky stuff? :-P

Reply:
To match TextWidget's signature, because Button might call it since the multi-line support pr ;).

(Also, I needed it too ;p).